### ISSUE TITLE
IRCClient reports ":None:None" to CTCP version req

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3660]
|<img alt="CaptSolo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| CaptSolo reported|
|-|-|
|Trac ID|trac#3660|
|Type|enhancement|
|Created|2009-02-18 18:39:24Z|
|Branch|https://github.com/twisted/twisted/tree/ctcp-version-3660-2|

IRCClient responds to CTCP version requests if the "versionName" variable is set. 

However, if variables "versionNum" and "versionEnv" are not set (documentation does not say they are mandatory, just says that they default to None) then text "None" is sent in the version reply:

```
CTCP VERSION reply from ...: versionName:None:None
```

Is that the intended behaviour? Would look better if they are omitted when None.

<details><summary>Searchable metadata</summary>

```
trac-id__3660 3660
type__enhancement enhancement
reporter__CaptSolo CaptSolo
priority__highest highest
milestone__ 
branch__branches_ctcp_version_3660_2 branches/ctcp-version-3660-2
branch_author__washort washort
status__closed closed
resolution__fixed fixed
component__words words
keywords__irc irc
time__1234982364000000 1234982364000000
changetime__1266327831000000 1266327831000000
version__None None
owner__washort washort

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="CaptSolo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|CaptSolo set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)|@washort removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @washort|
|-|-|

Where's the code?

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_4"></a>|@mwhudson commented|
|-|-|

In branches/ctcp-version-3643 ?

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_5"></a>|@washort commented|
|-|-|

(In [26266]) Branching to 'ctcp-version-3660'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @washort|
|-|-|

There is a small conflict with trunk apparently.

```
+            vstring = self.versionName
```

This variable looks unused.

Please merge once fixed!

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_8"></a>|@washort commented|
|-|-|

(In [26329]) Branching to 'ctcp-version-3660-2'

**Comment by twisted-trac** [comment]
|<img alt="Poincare's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|Poincare commented|
|-|-|

It doesn't matter what the answer if the vars needed aren't filled out because obviously the programmer doesn't care about the environment sent for CTCP requests.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [28215]) Bump copyright date; add news file; adjust docstrings and assertions to match convention

refs [#3660](https://github.com/twisted/twisted/issues/3660)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [28216]) Merge ctcp-version-3660-2

Author: washort
Reviewer: therve
Fixes: [#3660](https://github.com/twisted/twisted/issues/3660)

Fix IRCClient's VERSION responder to exclude version number and
environment if `versionNum` or `versionEnv` is set to None.
